psi4/src/psi4/fisapt/fisapt.cc

```diff
@@ -1096,6 +1158,7 @@ void FISAPT::dHF() {
     LH_BC->copy(T);
     LH_BC->add(V_B);
     LH_BC->add(V_C);
+    if (reference_->external_pot()) LH_BC->add(matrices_["VE"]);
```

Comment:
Ah, this kind of arithmetic makes the external potential tricky. If we added it to `V` we would need the following extra matrices:
```
V_AB
V_AC
V_BC
```

I am actually leaning a bit towards this as future perturbations would be simpler. Any thoughts here?

Reply:
This makes a lot of sense to me. It avoids a lot of unnecessary if statements. Should I change to doing things this way?

Reply:
I think so, lets do it this way unless there are any blockers. @psi4/admins

Reply:
On second thought, one problem with this approach is that the functions that compute the terms in the SAPT energy, e.g. elst(), also use the VA and VB matrices. If we add the external potential to VA and VB, it will be included in those SAPT energies. This seems to disagree with how the external potential is handled in Ed's SAPT module, which accounts for it in the HF energy through the SCF module but uses the molecule's one-electron potential without the external potential to compute the SAPT terms. Indeed, after making this change (I haven't committed it yet), the FISAPT module's total SAPT0 energy disagrees with the SAPT module's total SAPT0 energy. If including the external potential in the SAPT terms is the right thing to do, we also need to change this in the SAPT module. What do you think?

Reply:
I think you are right. It would be a good to have a central place, but this would generate even more intermediates. I don't suppose you see a better solution than the current code here?

Reply:
One option is to define V_AB, V_AC, and V_BC locally within the dHF() function. This would still require keeping matrices_["VE"] as a global matrix that's added to the appropriate potential matrix in the call to FISAPTSCF and within dHF(), but at least the change would be in one place in dHF().